http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Http2IT.java

```diff
@@ -0,0 +1,261 @@
+                .setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.httpClientBuilder().with(httpClientOptions).build();
+        httpClient.request(HttpMethod.GET, baseUri.getPort(), baseUri.getHost(), BASE_ENDPOINT)
+                .compose(request -> request.send()
+                        .compose(httpClientResponse -> {
+                            assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                            assertEquals(HttpVersion.HTTP_2, httpClientResponse.version());
+                            return httpClientResponse.body();
+                        }))
+                .onSuccess(body -> vertxTestContext.verify(() -> {
+                    assertThat("Body response", body.toString().contains(GREETING));
+                    vertxTestContext.completeNow();
+                })).onFailure(Throwable::printStackTrace);
+
+    }
+
+    @Test
+    @DisplayName("Test max header size exceeded error message")
+    @Order(5)
+    void testMaxHeaderSizeExceeded(VertxTestContext testContext) {
+        HttpClientOptions options = new HttpClientOptions()
+                .setSsl(true)
+                .setUseAlpn(true)
+                .setTrustOptions(new JksOptions().setPath("keystore.jks").setPassword("password"))
+                .setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.httpClientBuilder().with(options).build();
+
+        int expectedLimit = 200;
+        int headerCount;
+
+        for (int i = 0; i < expectedLimit; i++) {
+            headerCount = i;
+
+            try {
+                request(headerCount, testContext);
+            } catch (Exception e) {
+                if (e.getMessage().contains("Header size exceeded max allowed size")) {
+                    testContext.completeNow();
+                    break;
+                } else {
+                    testContext.failNow(e);
+                }
+            }
+        }
+    }
+
+    private static void request(int headerCount, VertxTestContext testContext) {
+
+        httpClient.request(HttpMethod.GET, app.getPort(Protocol.HTTPS), baseUri.getHost(), BASE_ENDPOINT)
+                .compose(request -> {
+                    for (int i = 0; i < headerCount; i++) {
+                        String headerName = "Header" + i;
+                        String headerValue = "Value" + i;
+                        request.putHeader(headerName, headerValue);
+                    }
+                    return request.send();
+                })
+                .onFailure(throwable -> {
+                    if (throwable instanceof Http2Exception &&
+                            throwable.getMessage().contains("Header size exceeded max allowed size ")) {
+                        testContext.completeNow();
+                    } else {
+                        testContext.failNow(throwable.getMessage() + headerCount);
```

Comment:
```suggestion
                        testContext.failNow(throwable.getMessage() + " ; header count is" + headerCount);
```
or some better text. the point is there is an exceptionmessage51 (where 51 will be headerCount => no space)